src/Data/Text/Builder/Linear/Char.hs

```diff
@@ -74,3 +90,112 @@ unsafePrependCharM marr off c = case utf8Length c of
     A.unsafeWrite marr (off - 2) n2
     A.unsafeWrite marr (off - 1) n3
     pure 4
+
+--------------------------------------------------------------------------------
+-- Multiple chars
+--------------------------------------------------------------------------------
+
+-- | Prepend a given count of a 'Char' to a 'Buffer'.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> prependChars 3 'x' (b |>. 'A'))
+-- "xxxA"
+prependChars ∷ Word → Char → Buffer ⊸ Buffer
+prependChars count ch buff
+  | count == 0 = buff
+  | isAscii ch =
+      prependExact
+        (fromIntegral count)
+        (\dst dstOff → unsafeReplicate dst dstOff (fromIntegral count) (ord ch))
+        buff
+  | otherwise =
+      case utf8Length ch of
+        cLen → case cLen * fromIntegral count of
+          totalLen →
+            prependExact
+              totalLen
+              (\dst dstOff → unsafeWrite dst dstOff ch *> unsafeTile dst dstOff totalLen cLen)
+              buff
```

Comment:
Let's reduce code duplication here: in both cases you can run `prependExact (utf8Length ch * fromIntegral count) fun buff`, it's only `fun` which depends on `isAscii ch`.

`utf8Length` is a fast function, there is little to save by not calling it.

Reply:
Done